src/Raven.Client/Documents/Linq/LinqPathProvider.cs

```diff
@@ -514,6 +520,12 @@ private static void AssertNoComputation(MemberExpression memberExpression)
                 switch (cur.Expression?.NodeType)
                 {
                     case ExpressionType.Call:
+                        if (cur.Expression is MethodCallExpression mce && mce.Method.Name == "get_Item")
+                        {
+                            break;
+                            //nested dictionary support eg. Where(i => W["x"]["y"].Name == "test")
+                        }
+                        goto ExceptionMsg;
```

Comment:
Let's try to avoid `goto`. You can introduce local method which throw which exception and call it in both places.